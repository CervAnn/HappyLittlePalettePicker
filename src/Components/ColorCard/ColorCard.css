.ColorCard {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  /* width: 100%; */
  /* height: 100%; */
}

#color-circle {
  border-radius: 100%;
  width: 200px;
  height: 200px;
  border: 1px solid black;
}

#lock-img {
  height: 25px;
  margin: 2px 2px 2px 2px;
}

#lock-img:hover {
  border: 1px solid #FFFFFF;
}