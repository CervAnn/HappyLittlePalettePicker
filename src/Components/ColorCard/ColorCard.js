import React from 'react'
import './ColorCard.css'
import PropTypes from 'prop-types';

const ColorCard = (props) => {
  return (
    <section className="ColorCard">
      <div id="color-circle" style={{backgroundColor: props.color}}></div>
      <p id="hex-code">{props.color}</p>
      <img id="lock-img"
      src={!props.isLocked ? require('../../assets/unlocked.svg') : require('../../assets/locked.svg')} 
      onClick={e => props.toggleLock(e, props.color)}
      alt=""/>
    </section>
  )
}

export default ColorCard;

ColorCard.propTypes = {
  color: PropTypes.string,
  delete: PropTypes.func,
  folderID: PropTypes.string,
  id: PropTypes.num,
  isLocked: PropTypes.bool,
  name: PropTypes.string,
  toggleLock: PropTypes.func
} 
