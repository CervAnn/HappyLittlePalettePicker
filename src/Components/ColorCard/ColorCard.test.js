import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import ColorCard from './ColorCard'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ColorCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ColorCard
    color={"#586BA4"}
    delete={jest.fn()}
    folderID={16558}
    id={114}
    isLocked={true}
    name={"Whale Life"}
    toggleLock={jest.fn()}
    />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleLock on a click', () => {
    const mockEvent = { preventDefault: jest.fn() }
    const mockToggleLock = jest.fn()

    wrapper = shallow(<ColorCard
      color={"#586BA4"}
      delete={jest.fn()}
      folderID={16558}
      id={114}
      isLocked={true}
      name={"Whale Life"}
      toggleLock={mockToggleLock}
      />);

    wrapper.find('#lock-img').simulate('click', mockEvent);

    expect(mockToggleLock).toHaveBeenCalled();
  });
})