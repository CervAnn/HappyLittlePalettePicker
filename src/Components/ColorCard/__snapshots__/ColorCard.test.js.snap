// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match the locked color snapshot 1`] = `
<section
  className="ColorCard"
>
  <div
    id="color-circle"
    style={
      Object {
        "backgroundColor": "#586BA4",
      }
    }
  />
  <p
    id="hex-code"
  >
    #586BA4
  </p>
  <img
    alt=""
    id="lock-img"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "locked.svg",
      }
    }
  />
</section>
`;

exports[`ColorCard should match the snapshot with all data passed in correctly 1`] = `
<section
  className="ColorCard"
>
  <div
    id="color-circle"
    style={
      Object {
        "backgroundColor": "#586BA4",
      }
    }
  />
  <p
    id="hex-code"
  >
    #586BA4
  </p>
  <img
    alt=""
    id="lock-img"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "locked.svg",
      }
    }
  />
</section>
`;
